pkg/util/fs.go

```diff
@@ -77,39 +76,75 @@ func (h *fs) Exists(file string) bool {
 	return err == nil
 }
 
-// Copy copies a set of files from sourcePath to targetPath
-func (h *fs) Copy(sourcePath string, targetPath string) error {
-	if _, err := os.Stat(sourcePath); err != nil {
+// Copy copies the source file to a destination directory.
+// If the destination is a filename, the file will be replaced.
+// If the source is directory, the directory itself and also the content of the
+// source directory will be copied into a target directory.
+func (h *fs) Copy(source string, dest string) (err error) {
+	sourcefile, err := os.Open(source)
+	if err != nil {
 		return err
 	}
+	sourceinfo, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+	defer sourcefile.Close()
 
-	info, err := os.Stat(targetPath)
-
-	if err != nil || (info != nil && !info.IsDir()) {
-		err = os.Mkdir(targetPath, 0700)
+	if sourceinfo.IsDir() {
+		glog.V(5).Infof("D %q -> %q", source, dest)
+		err = os.MkdirAll(dest, sourceinfo.Mode())
 		if err != nil {
 			return err
 		}
+		return h.CopyContents(source, dest)
+	}
+
+	destfile, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer destfile.Close()
+	glog.V(5).Infof("F %q -> %q", source, dest)
+	if _, err := io.Copy(destfile, sourcefile); err != nil {
+		return err
+	}
 
-		targetPath = filepath.Join(targetPath, filepath.Base(sourcePath))
+	if runtime.GOOS == "windows" {
+		return
 	}
-	// TODO: Use the appropriate command for Windows
-	glog.V(5).Infof("cp -a %s %s", sourcePath, targetPath)
-	return h.runner.Run("cp", "-a", sourcePath, targetPath)
+	return os.Chmod(dest, sourceinfo.Mode())
 }
 
-func (h *fs) CopyContents(sourcePath string, targetPath string) error {
-	info, err := os.Stat(sourcePath)
+// CopyContents copies the files recursively from the source directory to
+// destination directory.
+// The source directory itself is not copied, only the content of the source
+// directory.
+func (h *fs) CopyContents(src string, dest string) (err error) {
+	sourceinfo, err := os.Stat(src)
 	if err != nil {
 		return err
 	}
-	if !info.IsDir() {
-		return fmt.Errorf("source path %s is not a directory", sourcePath)
+	if err := os.MkdirAll(dest, sourceinfo.Mode()); err != nil {
+		return err
+	}
+	directory, err := os.Open(src)
+	if err != nil {
+		return err
 	}
-	if !strings.HasSuffix(sourcePath, string(filepath.Separator)) {
-		sourcePath += string(filepath.Separator)
+	objects, err := directory.Readdir(-1)
+	if err != nil {
+		return err
+	}
+	for _, obj := range objects {
+		source := path.Join(src, "/", obj.Name())
+		destination := path.Join(dest, "/", obj.Name())
```

Comment:
@PI-Victor err filepath.Join does that, here we always want to have "/" as this run in docker container which is always unix